src/lib/hash/keccak_fips/keccak_fips.h

```diff
@@ -0,0 +1,118 @@
+/*
+* Keccak-FIPS
+* (C) 2010,2016 Jack Lloyd
+* (C) 2023 Falko Strenzke
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_KECCAK_FIPS_H_
+#define BOTAN_KECCAK_FIPS_H_
+
+#include <botan/hash.h>
+#include <botan/secmem.h>
+#include <string>
+
+namespace Botan {
+
+/**
+* KECCAK FIPS
+*/
+
+class Keccak_FIPS_generic : public HashFunction
```

Comment:
Generic idea to consider: Would it make sense to define this as a template with the capacity being a value template parameter? Downstream users of the static building blocks could then do `Keccak_FIPS_256::absorb()` without explicitly stating the bitrate every time.

Not sure whether that's at all feasible, but maybe its worth a try in a follow-up.

Reply:
I had the same thought but refrained from trying it right now because I wasn't sure how much code would then have to go into the header and what would be the consequences of that. Right now I think it wouldn't be a problem as the header is not part of the public API. Making the capacity (and maybe other parameters) template parameters may improve the readability of the code and - at least potentially – increase performance.

However I would tend to first have the basic implementation merged and then address this later like you are proposing as well.